src/Workspaces/Core/Portable/SymbolId/SymbolKey.cs

```diff
@@ -225,8 +226,10 @@ private static SymbolKeyResolution CreateSymbolInfo(ISymbol[] symbols)
 
             for (int i = 0; i < refKinds.Length; i++)
             {
+                // The ref-out distinction is not interesting for SymbolKey because you can't overload
+                // based on the difference.
                 var parameter = parameters[i];
-                if (refKinds[i] != parameters[i].RefKind)
+                if (!SymbolEquivalenceComparer.AreEquivalent(refKinds[i], parameter.RefKind, distinguishRefFromOut: false))
```

Comment:
Might be best to name your new method AreRefKindsEquivalent...I read this diff as you're invoking the entire symbol asking if two symbols are equivalent.